tests/rust_no_std_compilation_test/src/main.rs

```diff
@@ -0,0 +1,35 @@
+#![no_std]
+#![no_main]
+#![feature(default_alloc_error_handler)]
+
+
+use core::alloc::{GlobalAlloc, Layout};
+
+struct NullAllocator;
+unsafe impl GlobalAlloc for NullAllocator {
+    unsafe fn alloc(&self, _lt: Layout) -> *mut u8 {
+        core::ptr::null_mut()
+    }
+    unsafe fn dealloc(&self, _ptr: *mut u8, _lt: Layout) {
+        panic!("won't deallocate: we never allocated!");
+    }
+}
+
+#[global_allocator]
+static A: NullAllocator = NullAllocator;
+
+
+#[panic_handler]
+fn panic(_info: &core::panic::PanicInfo) -> ! {
+    loop {}
+}
+
+
+// Include flatbuffers purely to check that it compiles in a no_std binary
```

Comment:
nit: include this at the top of the file (and maybe add a comment saying everything below is boilerplate for a null program)